src/TestLisExpr.chpl

```diff
@@ -62,19 +62,109 @@ module TestLisExpr
         return f;
     }
 
+     // tasks per locale based on locale0
+    const numTasks = here.maxTaskPar;
+    const Tasks = {0..#numTasks}; // these need to be const for comms/performance reasons
+    
+    // calculate sub-domain for task
+    inline proc calcBlock(task: int, low: int, high: int) {
+        var totalsize = high - low + 1;
+        var div = totalsize / numTasks;
+        var rem = totalsize % numTasks;
+        var rlow: int;
+        var rhigh: int;
+        if (task < rem) {
+            rlow = task * (div+1) + low;
+            rhigh = rlow + div;
+        }
+        else {
+            rlow = task * div + rem + low;
+            rhigh = rlow + div - 1;
+        }
+        return {rlow .. rhigh};
+    }
+
+    /* eval with a specific environment for the test expression return true if no errors */
+    proc test_eval_coforall(const in prog: string): bool {
+        var f = true;
+        var N = 10;
+        var D = {0..#N};
+        var A: [D] int = D;
+        var B: [D] int;
+        
+        // this could have the advantage of not creating array temps like the rest of arkouda does
+        try {
+            coforall loc in Locales {
+                on loc {
+                    coforall task in Tasks {
+                        // get local domain's indices
+                        var lD = A.domain.localSubdomain();
+                        // calc task's indices from local domain's indices
+                        var tD = calcBlock(task, lD.low, lD.high);
+                        
+                        // parse and check the program
+                        var ast = parse(prog);
+                        // allocate the env for variables
+                        var env = new owned Env();
+                        
+                        for i in tD {
+                            // addEnrtry redefines values for already existing entries
+                            env.addEntry("elt",A[i]); // add a symbol called "elt" and value for a
+                            
+                            // this version does the eval the in the enviroment which creates the symbol "ans"
+                            //var ans = env.lookup("ans").toValue(int).v; // retrieve value for ans
+                            //b = ans;
+                            
+                            // this version just returns the GenValue from the eval call
+                            var ans = eval(ast,env);
+                            B[i] = ans.toValue(int).v; // put answer into b
+                        }
+                    }
+                }
+                
+                writeln(A);
+                writeln(B);
+            }
+        }
+        catch errors: TaskErrors {
+            for e in errors {
+                // why does compile complain about not having a try her and not in the serial case???!
+                try! writeln((e: unmanaged Error).message());
```

Comment:
@bradcray @bmcdonald3 for some reason I had to cast my `owned Error` to `unmanaged Error` to get this to work.

Reply:
Hmm, similarly, I did not need the unmanaged cast on 1.26. Was this a compilation error that you were seeing?

Reply:
here is the error I got
```
TestLisExpr.chpl:88: In function 'test_eval_coforall':
TestLisExpr.chpl:131: error: call to throwing cast without throws, try, or try! (relaxed mode)
```

Reply:
here is the other error I got before I used the cast
```
TestLisExpr.chpl:88: In function 'test_eval_coforall':
TestLisExpr.chpl:131: error: unresolved call 'type Error?.message()'
$CHPL_HOME/modules/standard/Errors.chpl:72: note: this candidate did not match: Error.message()
TestLisExpr.chpl:131: note: because method call receiver is a type
$CHPL_HOME/modules/standard/Errors.chpl:72: note: but is passed to non-type formal 'this: borrowed Error'
TestLisExpr.chpl:131: note: other candidates are:
$CHPL_HOME/modules/standard/Errors.chpl:83: note:   NilThrownError.message()
$CHPL_HOME/modules/standard/Errors.chpl:90: note:   NilClassError.message()
note: and 3 other candidates, use --print-all-candidates to see them
```

Reply:
OK, I think that I understand a bit better now. You are seeing the error for the `try!` not from the `writeln`, but from the cast to an `unmanaged Error`. The second error you are seeing: `error: unresolved call 'type Error?.message()'`, comes from the error being nilable, so you need to end it with a `!` in order to call the error message. The reason that the cast to `unmanaged Error` fixed it is because you are casting from a nilable to a non-nilable object, but I think it is more sidestepping the problem rather than addressing it.

I would recommend doing something like this diff:
```diff
diff --git a/src/TestLisExpr.chpl b/src/TestLisExpr.chpl
index e5caa51..c1a54ff 100644
--- a/src/TestLisExpr.chpl
+++ b/src/TestLisExpr.chpl
@@ -126,18 +126,16 @@ module TestLisExpr
                 writeln(B);
             }
         }
-        catch errors: TaskErrors {
-            for e in errors {
-                // why does compile complain about not having a try her and not in the serial case???!
-                try! writeln((e: unmanaged Error).message());
+        catch e: Error {
+          if e.type == TaskErrors {
+            for err in e {
+              writeln(err.message());
             }
-            f = false;
-        }
-        catch {
-            try! writeln("unkown exception!!!");
-            f = false;
+          } else {
+            writeln(e.message());
+          }
+          f = false;
         }
-

         return f;
     }
```

Where we switch to catching a generic `Error` and then checking the type, which removes the need for the nil checking, the `try!` on the writeln, and the extra generic `catch` that was added.

Do you feel better about this approach? If not, it would be helpful to know what you were looking for here so I can be a little bit more helpful.